pkg/controller/ignition/controller.go

```diff
@@ -0,0 +1,216 @@
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type Controller struct {
+	*controllertools.Observer
+
+	namespace                   string
+	serviceName                 string
+	nodesBroadcastAddressType   scyllav1.BroadcastAddressType
+	clientsBroadcastAddressType scyllav1.BroadcastAddressType
+
+	ignited atomic.Bool
+
+	configMapLister corev1listers.ConfigMapLister
+	serviceLister   corev1listers.ServiceLister
+	podLister       corev1listers.PodLister
+}
+
+func NewController(
+	namespace string,
+	serviceName string,
+	clientsBroadcastAddressType scyllav1.BroadcastAddressType,
+	nodesBroadcastAddressType scyllav1.BroadcastAddressType,
+	kubeClient kubernetes.Interface,
+	configMapInformer corev1informers.ConfigMapInformer,
+	serviceInformer corev1informers.ServiceInformer,
+	podInformer corev1informers.PodInformer,
+) (*Controller, error) {
+	controller := &Controller{
+		namespace:                   namespace,
+		serviceName:                 serviceName,
+		clientsBroadcastAddressType: clientsBroadcastAddressType,
+		nodesBroadcastAddressType:   nodesBroadcastAddressType,
+		ignited:                     atomic.Bool{},
+		configMapLister:             configMapInformer.Lister(),
+		serviceLister:               serviceInformer.Lister(),
+		podLister:                   podInformer.Lister(),
+	}
+
+	observer := controllertools.NewObserver(
+		"scylladb-ignition",
+		kubeClient.CoreV1().Events(corev1.NamespaceAll),
+		controller.Sync,
+	)
+
+	configMapHandler, err := configMapInformer.Informer().AddEventHandler(observer.GetGenericHandlers())
+	if err != nil {
+		return nil, fmt.Errorf("can't add ConfigMap event handler: %w", err)
+	}
+	observer.AddCachesToSync(configMapHandler.HasSynced)
+
+	serviceHandler, err := serviceInformer.Informer().AddEventHandler(observer.GetGenericHandlers())
+	if err != nil {
+		return nil, fmt.Errorf("can't add Service event handler: %w", err)
+	}
+	observer.AddCachesToSync(serviceHandler.HasSynced)
+
+	podHandler, err := podInformer.Informer().AddEventHandler(observer.GetGenericHandlers())
+	if err != nil {
+		return nil, fmt.Errorf("can't add Pod event handler: %w", err)
+	}
+	observer.AddCachesToSync(podHandler.HasSynced)
+
+	controller.Observer = observer
+
+	return controller, nil
+}
+
+func (c *Controller) IsIgnited() bool {
+	return c.ignited.Load()
+}
+
+func (c *Controller) Sync(ctx context.Context) error {
+	startTime := time.Now()
+	klog.V(4).InfoS("Started syncing observer", "Name", c.Observer.Name(), "startTime", startTime)
+	defer func() {
+		klog.V(4).InfoS("Finished syncing observer", "Name", c.Observer.Name(), "duration", time.Since(startTime))
+	}()
+
+	ignited := true
+
+	svc, err := c.serviceLister.Services(c.namespace).Get(c.serviceName)
+	if err != nil {
+		return fmt.Errorf("can't get service %q: %w", c.serviceName, err)
+	}
+
+	// TODO: This isn't bound to the lifecycle of the ScyllaDB Pod and should be evaluated in the controller.
```

Comment:
leftover? It's part of the controller now?

Reply:
nope, but I lost the reference to the issue https://github.com/scylladb/scylla-operator/issues/604

Reply:
added the issue link